File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2905-find-indices-with-index-and-value-difference-ii/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2905-find-indices-with-index-and-value-difference-ii.py
class Solution:
    def findIndices(self, nums: List[int], indexDifference: int, valueDifference: int):
        mx=-1
        mn=-1
        
        for i,num in enumerate(nums):
            if i-indexDifference>=0:
                if mn==-1 or nums[i-indexDifference]<nums[mn]:
                    mn = i-indexDifference
                if mx==-1 or nums[i-indexDifference]>nums[mx]:
                    mx = i-indexDifference
                if num-nums[mn]>= valueDifference:
                    return [mn,i]
                if nums[mx]-num>=valueDifference:
                    return [mx,i]
        return [-1,-1]

File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2905-find-indices-with-index-and-value-difference-ii/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2905-find-indices-with-index-and-value-difference-ii.cpp
class Solution {
public:
    vector<int> findIndices(vector<int>& nums, int indexDifference, int valueDifference) {
        int mx,mn;
        mx=-1;
        mn=-1;
        
        int n = nums.size();
        vector<int> ans(2,-1);
        
        for (int i=0;i<n;i++){
            if (i>=indexDifference){
                if(mn==-1 || nums[i-indexDifference]<nums[mn]) {
                    mn = i-indexDifference;
                }
                
                if(mx==-1 || nums[i-indexDifference]>nums[mx]) {
                    mx = i - indexDifference;}
                
                
                if ((nums[i]-nums[mn])>=valueDifference){
                    ans[0]=mn;
                    ans[1]=i;
                    return ans;}
                if ((nums[mx]-nums[i])>=valueDifference){
                    ans[0]=i;
                    ans[1]=mx;
                    return ans;
                }
                
            }
        }
        return {-1,-1};
        
    }
};